carrier.go

```diff
@@ -50,34 +48,59 @@ type CarrierType struct {
 }
 
 type carrierAccountRequest struct {
-	CarrierAccount *CarrierAccount `json:"carrier_account,omitempty"`
+	Data *CarrierAccount `json:"carrier_account,omitempty"`
 }
 
-func (c *Client) selectCarrierAccountCreationEndpoint(account CarrierAccount) string {
+// UpsCarrierAccountCreationParameters contains the parameters needed to create a new UPS carrier account.
+type UpsCarrierAccountCreationParameters struct {
+	Type          string `json:"type,omitempty"`
+	Description   string `json:"description,omitempty"`
+	Reference     string `json:"reference,omitempty"`
+	AccountNumber string `json:"account_number,omitempty"`
+}
+
+type upsCarrierAccountCreationRequest struct {
+	Data *UpsCarrierAccountCreationParameters `json:"ups_oauth_registrations,omitempty"`
+}
+
+// UpsCarrierAccountUpdateParameters contains the parameters needed to update a UPS carrier account.
+type UpsCarrierAccountUpdateParameters struct {
+	// TODO: Add Description and Reference if/when API allows updating them.
+	AccountNumber string `json:"account_number,omitempty"`
+}
+
+type upsCarrierAccountUpdateRequest struct {
+	Data *UpsCarrierAccountUpdateParameters `json:"ups_oauth_registrations,omitempty"`
+}
+
+func (c *Client) selectCarrierAccountCreationEndpoint(typ string) string {
 	for _, carrier := range getCarrierAccountTypesWithCustomWorkflows() {
-		if account.Type == carrier {
+		if typ == carrier {
```

Comment:
question: is `type` a builtin for Go?

Reply:
Yes, it will complain if you use it. Just like "klass" in Java, lol

<img width="902" alt="image" src="https://github.com/EasyPost/easypost-go/assets/17054780/a6ae692c-04a4-45e4-9d11-45f17556f86b">